src/sql/base/browser/ui/infoButton/infoButton.ts

```diff
@@ -0,0 +1,183 @@
+			this._textContainer.style.color = '#006ab1';
+			this._textContainer.style.display = 'flex';
+			this._textContainer.style.flexFlow = 'column';
+			this._textContainer.style.justifyContent = 'space-between';
+			this._textContainer.style.padding = '0 0 0 10px';
+			this._textContainer.style.margin = '0px';
+
+			this._pTitle = document.createElement('p');
+			this._pTitle.setAttribute('aria-hidden', 'false');
+			this._pTitle.style.fontSize = '14px';
+			this._pTitle.style.fontWeight = 'bold';
+			this._pTitle.style.lineHeight = '20px';
+			this._pTitle.style.margin = '0px';
+
+			this._pDesc = document.createElement('p');
+			this._pDesc.setAttribute('aria-hidden', 'false');
+			this._pDesc.style.fontSize = '12px';
+			this._pDesc.style.lineHeight = '16px';
+			this._pDesc.style.margin = '0px';
+
+			this._textContainer.appendChild(this._pTitle);
+			this._textContainer.appendChild(this._pDesc);
+
+			this._iconContainer.appendChild(this._iconElement);
+
+			this._main.appendChild(this._iconContainer);
+			this._main.appendChild(this._textContainer);
+			this.element.appendChild(this._main);
+			this.element.style.background = 'none';
+			this.element.style.display = 'inline-block';
+		}
+		this.infoButtonOptions = options;
+	}
+
+	public get textTitle(): string {
+		return this._textTitle;
+	}
+	public set textTitle(value: string) {
+		this._textTitle = value;
+		this._pTitle.innerText = this.textTitle;
+	}
+
+	public get description(): string {
+		return this._description;
+	}
+	public set description(value: string) {
+		this._description = value;
+		this._pDesc.innerText = this.description;
+	}
+
+	public get buttonMaxHeight(): number {
+		return this._buttonMaxHeight;
+	}
+	public set buttonMaxHeight(value: number) {
+		this._buttonMaxHeight = value;
+		this._main.style.maxHeight = this._buttonMaxHeight.toString() + 'px';
+		this._iconContainer.style.height = (this._buttonMaxHeight - 20).toString() + 'px';
+		this._textContainer.style.height = (this._buttonMaxHeight - 20).toString() + 'px';
+	}
+
+	public get buttonMaxWidth(): number {
+		return this._buttonMaxWidth;
+	}
+	public set buttonMaxWidth(value: number) {
+		this._buttonMaxWidth = value;
+		this._main.style.width = this._buttonMaxWidth.toString() + 'px';
+		this._textContainer.style.width = (this._buttonMaxWidth - this._iconWidth).toString() + 'px';
+	}
+
+	public get iconHeight(): number {
+		return this._iconHeight;
+	}
+	public set iconHeight(value: number) {
+		this._iconHeight = value;
+		this._iconElement.style.height = this._iconHeight.toString() + 'px';
+	}
+
+	public get iconWidth(): number {
+		return this._iconWidth;
+	}
+	public set iconWidth(value: number) {
+		this._iconWidth = value;
+		this._iconContainer.style.width = this._iconWidth.toString() + 'px';
+		this._iconElement.style.width = this._iconWidth.toString() + 'px';
+	}
+
+	public get iconClass(): string {
+		return this._iconClass;
+	}
+	public set iconClass(value: string) {
+		this._iconClass = value;
```

Comment:
setting icon class won't do anything after initializing, Maybe updating icon is not a requirement for your scenario, but since it is very easy to implement, I'd add it.

Reply:
I'm using iconClass here to supply the CSS class corresponding to each button's icon. Removed iconPath from this context.